rewrite-test/src/main/kotlin/org/openrewrite/java/search/SemanticallyEqualTest.kt

```diff
@@ -0,0 +1,308 @@
+package org.openrewrite.java.search
+
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.Test
+import org.openrewrite.Formatting
+import org.openrewrite.Tree.randomId
+import org.openrewrite.java.JavaParser
+import org.openrewrite.java.tree.J
+import org.openrewrite.java.tree.JavaType
+
+interface SemanticallyEqualTest {
+
+    @Test
+    fun annotationEquality(jp: JavaParser) {
+        val cu = jp.parse(
+                """
+                    @MyAnnotation(value = true, srcValue = "true")
+                    class A {}
+                """,
+                """
+                    @interface MyAnnotation { 
+                        boolean value();
+                        String srcValue(); 
+                    }
+                """
+        )
+
+        val annotation = cu[0].classes[0].annotations[0]
+
+        assertThat(SemanticallyEqual(annotation)
+            .visit(
```

Comment:
While there's absolutely nothing wrong with manually constructing the J.Annotation object you're comparing against, there wouldn't be anything wrong with using a `JavaParser` to construct it and that probably would have been easier and faster for you.